lib/field/field.spec.tsx

```diff
@@ -1127,3 +1127,61 @@ test("Field should remove its value when unrendered", async () => {
     `)
   );
 });
+
+test("Field should not remove its value if preserveValue", async () => {
+  const Comp = () => {
+    const [show, setShow] = useState(true);
+    const [values, setValues] = useState<string | null>(null);
+
+    if (values) return <p>{values}</p>;
+
+    return (
+      <Form onSubmit={(values) => setValues(JSON.stringify(values))}>
+        {({ submit }) => (
+          <div>
+            <button onClick={() => setShow(false)}>Unmount</button>
+            <button onClick={submit}>Submit</button>
+            {show && (
+              <Field<string> name={"email"} initialValue="" preserveValue>
+                {({ value, setValue }) => (
+                  <input
+                    placeholder="Email"
+                    value={value}
+                    onChange={(e) => setValue(e.target.value)}
+                  />
+                )}
+              </Field>
+            )}
+            <Field<string> name={"password"} initialValue="">
+              {({ value, setValue }) => (
+                <input
+                  placeholder="Password"
+                  value={value}
+                  onChange={(e) => setValue(e.target.value)}
+                />
+              )}
+            </Field>
+          </div>
+        )}
+      </Form>
+    );
+  };
+
+  const { getByPlaceholderText, getByText, container } = render(<Comp />);
+
+  await user.type(getByPlaceholderText("Email"), "emailHere");
+  await user.type(getByPlaceholderText("Password"), "passwordHere");
+
+  await user.click(getByText("Unmount"));
+  await user.click(getByText("Submit"));
+
+  await waitFor(() =>
+    expect(container).toMatchInlineSnapshot(`
+      <div>
+        <p>
+          {"email":"emailHere","password":"passwordHere"}
+        </p>
+      </div>
+    `)
+  );
+});
```

Comment:
Before we merge, let's add another test that copies the above, but then `Remounts` the `Field` using `show != show` and watch the value persist

Reply:
I didn't get how to do this